qiskit/quantum_info/operators/mixins/multiply.py

```diff
@@ -0,0 +1,52 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2017, 2020.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""
+Mixin for operator scalar multiplication interface.
+"""
+
+from abc import ABC, abstractmethod
+
+
+class MultiplyMixin(ABC):
+    """Abstract Mixin for scalar multiplication.
+
+    This class defines the following operator overloads:
+
+        - ``*`` / ``__rmul__`
+        - ``/`` / ``__truediv__``
+        - ``__neg__``
```

Comment:
Is `__mul__` needed here?

Reply:
Current convention in master/stable is `__mul__` is used for dot: `3 * op` is ok, `op1 * op2` is ok `op * 3` is an error.

I think we should deprecate the current behaviour of `__mul__` and `__matmul__` for operators, but maybe that should be done in a separate PR